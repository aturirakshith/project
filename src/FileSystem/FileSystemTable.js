import getResultObj from "./FolderObject";
import React, { useState } from "react";
import Table from "@mui/material/Table";
import TableBody from "@mui/material/TableBody";
import TableCell from "@mui/material/TableCell";
import TableContainer from "@mui/material/TableContainer";
import TableHead from "@mui/material/TableHead";
import TableRow from "@mui/material/TableRow";
import tempData from "../Data";
import Paper from "@mui/material/Paper";
import Button from "@mui/material/Button";

export default function BasicTable(props) {
  let data2 = [
    {
      size: "1101",
      name: "src/folder1/folder2/20220413-16-28-15/correctcsv.csv",
      dlp_status: "scan-good1",
      created_date: "2022 04 13 16:28:16",
      type: null,
      last_modified: "2022 04 13 16:28:12",
    },
    {
      size: "1102",
      name: "src/folder1/folder2/20220413-16-28-15/correctcsv2.csv",
      dlp_status: "scan-good2",
      created_date: "2022 04 13 16:28:17",
      type: null,
      last_modified: "2022 04 13 16:28:13",
    },
    {
      size: "1103",
      name: "src/folder1/folder2/20220413-16-28-15/correctcsv3.csv",
      dlp_status: "scan-good3",
      created_date: "2022 04 13 16:28:18",
      type: null,
      last_modified: "2022 04 13 16:28:14",
    },
    {
      size: "1104",
      name: "src/folder1/folder2/20220413-16-28-15/correctcsv4.csv",
      dlp_status: "scan-good4",
      created_date: "2022 04 13 16:28:19",
      type: null,
      last_modified: "2022 04 13 16:28:15",
    },
    {
      size: "1105",
      name: "src/folder1/folder3/20220413-16-28-15/correctcsv.csv",
      dlp_status: "scan-good5",
      created_date: "2022 04 13 16:28:15",
      type: null,
      last_modified: "2022 04 13 16:28:16",
    },
    {
      size: "1106",
      name: "src/folder1/folder3/20220413-16-28-15/correctcsv2.csv",
      dlp_status: "scan-good6",
      created_date: "2022 04 13 16:28:16",
      type: null,
      last_modified: "2022 04 13 16:28:17",
    },
    {
      size: "1107",
      name: "src/folder1/20220413-16-28-15/correctcsv.csv",
      dlp_status: "scan-good7",
      created_date: "2022 04 13 16:28:17",
      type: null,
      last_modified: "2022 04 13 16:28:18",
    },
  ];

  const {
    details,
    datasetsList,
    statusClicked,
    uploadedfileType,
    uploadOrDownloadProgress,
  } = props;
  let data;
  if (datasetsList) {
    data = getResultObj(
      datasetsList,
      details,
      uploadedfileType,
      uploadOrDownloadProgress
    );
  } else {
    // will be removing this logic once the API starts sending proper data, for now using hardcoded for demo purposes.
    data = getResultObj(
      data2,
      details,
      uploadedfileType,
      uploadOrDownloadProgress
    );
  }
  const [keys, setKeys] = useState(Object.keys(data));
  const [values, setValues] = useState(Object.values(data));
  const [array, setArray] = useState([]);
  const headerdata = [
    {
      title: "Name",
      width: 20,
    },
    {
      title: "Type",
      width: 15,
    },
    {
      title: "Size",
      width: 15,
    },
    {
      title: "Creation Date",
      width: 20,
    },
    {
      title: "Last Modified",
      width: 20,
    },
    {
      title: "Status",
      width: 20,
    },
  ];

  const linkStyle = {
    border: "none",
    background: "none",
    textDecoration: "underline",
    cursor: "pointer",
    outline: "none",
    "&.hover": { color: "blue" },
  };

  const newObj = (array) => {
    let obj = data;
    for (const item of array) {
      obj = obj[item]["contents"];
    }
    return obj;
  };
  const changeState = (array, keys, values) => {
    setValues(values);
    setKeys(keys);
    setArray(array);
  };
  const handleClick = (value) => {
    const newArray = [...array, value];
    const obj = newObj(newArray);
    changeState(newArray, Object.keys(obj), Object.values(obj));
  };
  const handleBreadClick = (index) => {
    const newArray = array.slice(0, index + 1);
    const obj = newObj(newArray);
    changeState(newArray, Object.keys(obj), Object.values(obj));
  };
  const handleHome = (index) => {
    const newArray = array.slice(0, index);
    const obj = newObj(newArray);
    changeState(newArray, Object.keys(obj), Object.values(obj));
  };
  return (
    <>
      <span>
        <button
          className="breadcrumb-home-button"
          style={
            array.length == 0
              ? { ...linkStyle, textDecoration: "none" }
              : linkStyle
          }
          disabled={0 === array.length}
          onClick={() => handleHome(0)}
        >
          Home
        </button>
        {array.length !== 0 && <span> &gt; </span>}
      </span>

      {array.map((item, index) => {
        return (
          <span key={index}>
            <button
              className="breadcrumb-location-button"
              style={
                index === array.length - 1
                  ? { ...linkStyle, textDecoration: "none" }
                  : linkStyle
              }
              disabled={index === array.length - 1}
              onClick={() => handleBreadClick(index)}
            >
              {item}
            </button>
            {index !== array.length - 1 && <span> &gt; </span>}
          </span>
        );
      })}

      <TableContainer component={Paper}>
        <Table sx={{}} aria-label="simple table">
          <TableHead>
            <TableRow>
              {headerdata.map((data, index) => (
                <TableCell key={index} align="right">
                  {data.title}
                </TableCell>
              ))}
            </TableRow>
          </TableHead>
          <TableBody>
            {values.map((row, index) => (
              <TableRow
                key={index}
                sx={{ "&:last-child td, &:last-child th": { border: 0 } }}
              >
                <TableCell align="right">
                  {row.type == "folder" ? (
                    <button
                      className="breadcrumb-folder-button"
                      style={linkStyle}
                      onClick={() => handleClick(keys[index])}
                    >
                      {keys[index]}
                    </button>
                  ) : (
                    keys[index]
                  )}
                </TableCell>

                <TableCell align="right">{row.type}</TableCell>
                <TableCell align="right">{row.size}</TableCell>
                <TableCell align="right">{row.created_date}</TableCell>
                <TableCell align="right">{row.last_modified}</TableCell>
                <TableCell align="right">
                  <Button
                    className="status-filesystem"
                    style={{
                      //backgroundColor: row.buttonConfig.className,
                      color: "white",
                      width: "120px",
                    }}
                    onClick={() => statusClicked(row.buttonConfig.stage)}
                    id={`Status-${index}`}
                  ></Button>
                </TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    </>
  );
}
