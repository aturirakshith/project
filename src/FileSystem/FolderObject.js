const getValueByKey = (obj, value) => {
  let foundObject = null;
  Object.keys(obj).forEach((key) => {
    if (key === value) {
      foundObject = obj[key];
    } else {
      if (obj[key]["contents"]) {
        foundObject = getValueByKey(obj[key]["contents"], value);
      }
    }
  });
  return foundObject;
};

const ScanStatus = {
  SCAN_GOOD: "scan-good",
  UPLOAD_COMPLETE: "upload-complete",
  STARTED: "started",
  REVIEW_IN_PROGRESS: "review-in-progress",
  REVIEW_DENIED: "review-denied",
  REVIEW_APPROVED: "review-approved",
  SCAN_BAD_ATTESTATION: "scan-bad-attestation",
  FILE_PURGED: "file-purged",
  AUTO_EXPIRED: "auto-expired",
};

const setButtonName = (details, dlp_status) => {
  let buttonConfig = {
    buttonName: "",
    className: "",
    id: "",
    disabled: false,
    stage: "",
  };

  const { isDataUploadComplete, hiveTableName, showUpload } = details;
  if (hiveTableName) {
    if (isDataUploadComplete === false) {
    } else if (isDataUploadComplete === true) {
      if (dlp_status === ScanStatus.SCAN_GOOD) {
        buttonConfig.buttonName = "Done";
        buttonConfig.stage = "done";
        buttonConfig.className = "#0A853D";
      }
    }
  } else {
    if (isDataUploadComplete === false) {
      if (dlp_status === undefined) {
        buttonConfig.buttonName = "Upload file";
        buttonConfig.stage = "uploadFile";
        buttonConfig.className = "#3668DA";
      } else if (dlp_status === ScanStatus.UPLOAD_COMPLETE) {
        buttonConfig.buttonName = "Scan to begin";
        buttonConfig.stage = "scanToBegin";
        buttonConfig.className = "#3668DA";
      } else if (dlp_status === ScanStatus.STARTED && showUpload === true) {
        buttonConfig.buttonName = "File upload taking longer time";
        buttonConfig.stage = "fileUploadTakingLongerTime";
        buttonConfig.className = "#c58d03";
      } else if (dlp_status === ScanStatus.STARTED) {
        buttonConfig.buttonName = "Scan started";
        buttonConfig.stage = "scanStarted";
        buttonConfig.className = "#3668DA";
      } else if (dlp_status === ScanStatus.REVIEW_IN_PROGRESS) {
        buttonConfig.buttonName = "Review in progress";
        buttonConfig.stage = "reviewInProgress";
        buttonConfig.className = "#3668DA";
      } else if (dlp_status === ScanStatus.REVIEW_DENIED) {
        buttonConfig.buttonName = "Review denied";
        buttonConfig.stage = "reviewDenied";
        buttonConfig.className = "#BE1324";
      } else if (dlp_status === ScanStatus.SCAN_BAD_ATTESTATION) {
        buttonConfig.buttonName = "Scan bad";
        buttonConfig.stage = "scanBad";
        buttonConfig.className = "#BE1324";
      } else if (dlp_status === ScanStatus.FILE_PURGED) {
        buttonConfig.buttonName = "File Purged";
        buttonConfig.stage = "filePurged";
        buttonConfig.className = "#BE1324";
      } else if (dlp_status === ScanStatus.AUTO_EXPIRED) {
        buttonConfig.buttonName = "Auto expired";
        buttonConfig.stage = "autoExpired";
        buttonConfig.className = "#BE1324";
      }
    } else if (isDataUploadComplete === true) {
      if (dlp_status === ScanStatus.SCAN_GOOD) {
        buttonConfig.buttonName = "Scan completed";
        buttonConfig.stage = "scanCompleted";
        buttonConfig.className = "#0A853D";
      } else if (dlp_status === ScanStatus.REVIEW_APPROVED) {
        buttonConfig.buttonName = "Review approved";
        buttonConfig.stage = "reviewApproved";
        buttonConfig.className = "#0A853D";
      }
    }
  }
  return buttonConfig;
};

const getResultObj = (
  data,
  details,
  uploadedfileType,
  uploadOrDownloadProgress
) => {
  if (!data) return null;
  let resultObj = {};
  data.forEach((item, index) => {
    const buttonObj = setButtonName(details, item.dlp_status);
    const folderArr = item.name.split("/");
    folderArr.forEach((folder, index) => {
      if (item.name.indexOf(".") > -1) {
        if (resultObj[folder]) {
        } else {
          if (index === folderArr.length - 1) {
            let foundObject = getValueByKey(resultObj, folderArr[index - 1]);
            if (foundObject) {
              const fileNameArr = folder.split(".");
              foundObject.contents[fileNameArr[0]] = {};
              foundObject.contents[fileNameArr[0]].type = fileNameArr[1]
                ? fileNameArr[1]
                : "folder";
              foundObject.contents[fileNameArr[0]].created_date =
                foundObject.created_date;
              foundObject.contents[fileNameArr[0]].size = foundObject.size;
              foundObject.contents[fileNameArr[0]].last_modified =
                foundObject.last_modified;
              foundObject.contents[fileNameArr[0]].buttonConfig =
                foundObject.buttonConfig;
            } else {
              const fileNameArr = folder.split(".");
              resultObj[fileNameArr[0]] = {};
              resultObj[fileNameArr[0]].type = fileNameArr[1]
                ? fileNameArr[1]
                : "folder";
              resultObj[fileNameArr[0]].created_date = item.created_date;
              resultObj[fileNameArr[0]].size = item.size;
              resultObj[fileNameArr[0]].last_modified = item.last_modified;
              resultObj[fileNameArr[0]].buttonConfig = buttonObj;
            }
          } else {
            if (index != 0) {
              let foundObject = getValueByKey(resultObj, folderArr[index - 1]);
              if (foundObject) {
                if (!foundObject.contents[folder]) {
                  foundObject.contents[folder] = {};
                  foundObject.contents[folder].type = "folder";
                  foundObject.contents[folder].contents = {};
                  foundObject.contents[folder].created_date = item.created_date;
                  foundObject.contents[folder].size = item.size;
                  foundObject.contents[folder].last_modified =
                    item.last_modified;
                  foundObject.contents[folder].buttonConfig = buttonObj;
                }
              } else {
                resultObj[folder] = {};
                resultObj[folder].type = "folder";
                resultObj[folder].contents = {};
                resultObj[folder].created_date = item.created_date;
                resultObj[folder].size = item.size;
                resultObj[folder].last_modified = item.last_modified;
                resultObj[folder].buttonConfig = buttonObj;
              }
            } else {
              resultObj[folder] = {};
              resultObj[folder].type = "folder";
              resultObj[folder].contents = {};
              resultObj[folder].created_date = item.created_date;
              resultObj[folder].size = item.size;
              resultObj[folder].last_modified = item.last_modified;
              resultObj[folder].buttonConfig = buttonObj;
            }
          }
        }
      }
      // } else {
      //     if (!resultObj[folder]) {
      //         resultObj[folder] = {};
      //         resultObj[folder].type = "folder";
      //         resultObj[folder].contents = {};
      //         resultObj[folder].created_date = item.created_date;
      //         resultObj[folder].size = item.size;
      //         resultObj[folder].last_modified = item.last_modified;
      //         resultObj[folder].dlp_status = item.dlp_status;
      //     }
      // }
    });
  });

  return resultObj;
};
export default getResultObj;
