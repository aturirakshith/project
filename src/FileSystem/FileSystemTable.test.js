import React from "react";
import { shallow, mount } from "enzyme";
import FileSystemTable from "./FileSystemTable";
import * as getResultObj from "./FolderObject";

let data2 = [
  {
    size: "1101",
    name: "src/folder1/folder2/20220413-16-28-15/correctcsv.csv",
    dlp_status: "scan-good1",
    created_date: "2022 04 13 16:28:16",
    type: null,
    last_modified: "2022 04 13 16:28:12",
  },
  {
    size: "1102",
    name: "src/folder1/folder2/20220413-16-28-15/correctcsv2.csv",
    dlp_status: "scan-good2",
    created_date: "2022 04 13 16:28:17",
    type: null,
    last_modified: "2022 04 13 16:28:13",
  },
  {
    size: "1103",
    name: "src/folder1/folder2/20220413-16-28-15/correctcsv3.csv",
    dlp_status: "scan-good3",
    created_date: "2022 04 13 16:28:18",
    type: null,
    last_modified: "2022 04 13 16:28:14",
  },
  {
    size: "1104",
    name: "src/folder1/folder2/20220413-16-28-15/correctcsv4.csv",
    dlp_status: "scan-good4",
    created_date: "2022 04 13 16:28:19",
    type: null,
    last_modified: "2022 04 13 16:28:15",
  },
  {
    size: "1105",
    name: "src/folder1/folder3/20220413-16-28-15/correctcsv.csv",
    dlp_status: "scan-good5",
    created_date: "2022 04 13 16:28:15",
    type: null,
    last_modified: "2022 04 13 16:28:16",
  },
  {
    size: "1106",
    name: "src/folder1/folder3/20220413-16-28-15/correctcsv2.csv",
    dlp_status: "scan-good6",
    created_date: "2022 04 13 16:28:16",
    type: null,
    last_modified: "2022 04 13 16:28:17",
  },
  {
    size: "1107",
    name: "src/folder1/20220413-16-28-15/correctcsv.csv",
    dlp_status: "scan-good7",
    created_date: "2022 04 13 16:28:17",
    type: null,
    last_modified: "2022 04 13 16:28:18",
  },
];

//  let objData=getResultObj(data2)
const mockStatusClicked = jest.fn(() => {});

let props = {
  datasetsList: data2,
  details: {
    isDataUploadComplete: true,
    hiveTableName: "authzuser11__team_test",
    showUpload: false,
  },
  uploadedfileType: "CSV",
  uploadOrDownloadProgress: [],
  statusClicked: mockStatusClicked,
};
const buttonConfig = {
  buttonName: "Done",
  className: "#0A853D",
  disabled: false,
  id: "df",
  stage: "done",
};
const resultObj = {
  rslt: {
    buttonConfig,
    contents: {
      component: {
        type: "csv",
        buttonConfig,
      },
    },
    fat: "100g",
    type: "folder",
  },
  file3: {
    buttonConfig,
    type: "csv",
  },
  data: {
    buttonConfig,
    contents: {
      folder1: {
        buttonConfig,
        type: "folder",
        contents: {
          newfile: {
            buttonConfig,
            type: "jpeg",
          },
        },
      },
    },
    type: "folder",
  },
};

//getResultObj.default = jest.fn(() => resultObj);
describe("<FileSystemTable /> Testing", () => {
  beforeEach(() => {
    getResultObj.default = jest.fn(() => resultObj);
    //console.log(getResultObj.default);
  });
  it(" 1st Folder Button click", () => {
    // getResultObj.default = jest.fn(() => resultObj);
    // console.log(getResultObj.default);
    const wrapper = shallow(<FileSystemTable {...props} />);
    expect(wrapper.find(".breadcrumb-folder-button").at(0).text()).toBe("rslt");
  });
  it("renders correctly", () => {
    const wrapper = shallow(<FileSystemTable {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it(" Home Button exists", () => {
    const wrapper = shallow(<FileSystemTable {...props} />);
    // wrapper.find(".breadcrumb-folder-button").simulate("click");

    expect(wrapper.find(".breadcrumb-home-button")).toHaveLength(1);
  });

  it(" Initial Home Button Disabled", () => {
    const wrapper = shallow(<FileSystemTable {...props} />);
    expect(wrapper.find("button").at(0).props("disabled")).toBeTruthy();
  });
  it(" Home Button click", () => {
    const wrapper = shallow(<FileSystemTable {...props} />);
    wrapper.find(".breadcrumb-home-button").simulate("click");
  });

  it(" Breadcrumb  Button check", () => {
    const wrapper = shallow(<FileSystemTable {...props} />);
    wrapper.find(".breadcrumb-folder-button").at(0).simulate("click");
    expect(wrapper.find(".breadcrumb-location-button").at(0).text()).toBe(
      "rslt"
    );
  });

  it(" Breadcrumb  Button click", () => {
    const wrapper = shallow(<FileSystemTable {...props} />);
    wrapper.find(".breadcrumb-folder-button").at(0).simulate("click");
    wrapper.find(".breadcrumb-location-button").simulate("click");
  });
  it(" Breadcrumb  Button disabled on its location", () => {
    const wrapper = shallow(<FileSystemTable {...props} />);
    wrapper.find(".breadcrumb-folder-button").at(0).simulate("click");
    expect(
      wrapper.find(".breadcrumb-location-button").at(0).props().disabled
    ).toBeTruthy();
  });

  it(" Status clicked", () => {
    //const statusClicked = () => {};
    const wrapper = shallow(<FileSystemTable {...props} />);
    wrapper.find(".status-filesystem").at(0).simulate("click");
    expect(mockStatusClicked).toHaveBeenCalled();
  });

  it("NO Data Passed", () => {
    const wrapper = shallow(
      <FileSystemTable {...{ ...props, datasetsList: null }} />
    );
  });
});
