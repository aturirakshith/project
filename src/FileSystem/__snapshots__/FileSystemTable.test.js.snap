// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileSystemTable /> Testing renders correctly 1`] = `ShallowWrapper {}`;
